ecleankernel/layout/blspec.py

```diff
@@ -60,6 +61,44 @@ def __init__(self,
         else:
             raise LayoutNotFound(f"/boot/[EFI/]{self.kernel_id} not found")
 
+    def append_kernel_files(self,
+                            ftype: typing.Container[KernelFileType],
+                            path: Path,
+                            kernels: dict,
+                            ver: str,
+                            module_dict: dict,
+                            exclusions: typing.Container[KernelFileType] = [],
+                            ) -> typing.List[Kernel]:
+        fobj = GenericFile(path, ftype)
+
+        try:
+            k = kernels[ver]  # Reuse this if it already exists
+        except KeyError:
+            # This is not a version we have seen before
+            k = Kernel(ver)
```

Comment:
This change is not clear to me. Are you assuming that there the same kernel version may be present in both layouts? Have you verified that EK can actually handle this case?

Reply:
Yes and yes. This is also the reason for the for-if-else loop a couple of lines down.

We also need this try-except for initrd and linux in the type 1 layout since we call this function for each file we find in the directory of one kernel version.

Note that I also adjusted the ......._EFI test to test for a kernel version in only type 1, a kernel version in only type 2 and a kernel version installed in both layouts. In addition I tested this on my system and for a given kernel version eclean-kernel successfully removes both the uki and the linux+initrd.

Reply:
> We also need this try-except for initrd and linux in the type 1 layout since we call this function for each file we find in the directory of one kernel version.

But why do you need to do that? With the new approach, it should be perfectly fine to construct `Kernel` object unconditionally, and append files to that.

Reply:
As it is now it calls the function for each file, so both for inird and for linux. I can rewrite it so the construction of the kernel object is outside of that function but this would be slightly more code duplication which I was trying to avoid earlier.

Reply:
Fixed